packages/x-components/src/components/modals/base-id-modal-close.vue

```diff
@@ -1,36 +1,44 @@
 <template>
-  <BaseEventButton
-    v-on="$listeners"
-    :events="events"
-    class="x-button x-events-modal-id-close-button"
-    data-test="close-modal-id"
-  >
-    <!-- @slot (Required) Button content with a text, an icon or both -->
-    <slot />
-  </BaseEventButton>
+  <NoElement v-on="$listeners" data-test="close-modal-id">
+    <!--
+      @slot closer-element. It's the element that will trigger the modal closing. It's a
+      button by default.
+        @binding {Function} closeModal - The function to close the modal.
+    -->
+    <slot :closeModal="emitCloseModalEvent" name="closer-element">
+      <BaseEventButton
+        v-on="$listeners"
+        :events="{ UserClickedCloseModal: modalId }"
```

Comment:
This should be a normal button, you are not "using" the BaseEventButton.

`<button v-on="$listeners" @click="emitCloseModalEvent" class="x-button x-events-modal-id-close-button">`

If you do it like that, you can remove the line 11th, because you will emit that event in the click,.